python/paddle/fluid/incubate/fleet/parameter_server/pslib/__init__.py

```diff
@@ -0,0 +1,275 @@
+                    self._opt_info["fleet_desc"])
+                self._dist_desc = self._opt_info["fleet_desc"]
+            else:
+                print("You should run DistributedOptimizer.minimize() first")
+                sys.exit(-1)
+            self._fleet_ptr.init_server(self._dist_desc_str,
+                                        self.role_maker_._get_rank())
+            self.local_ip_ = self._fleet_ptr.run_server()
+            # barrier_all for init_server
+            self.role_maker_._barrier_all()
+            self.all_ips_ = self.role_maker_._all_gather(self.local_ip_)
+
+            self._fleet_ptr.gather_servers(self.all_ips_,
+                                           self.role_maker_._get_size())
+            # barrier_all for init_worker, wait all workers start
+            self.role_maker_._barrier_all()
+        else:
+            print("You should run DistributedOptimizer.minimize() first")
+            sys.exit(-1)
+
+    def stop_worker(self):
+        """
+        stop(): will be called after a user finishes his/her training task. Fleet instance will be
+            destroyed when stop() is called.
+        """
+        self.role_maker_._barrier_worker()
+        if self.role_maker_._is_first_worker():
+            self._fleet_ptr.stop_server()
+        self.role_maker_._barrier_worker()
+        self.role_maker_._barrier_all()
+        self.role_maker_._finalize()
+
+    def stop(self, executor):
+        """
+        stop(): will be called after a user finishes his/her training task. Fleet instance will be
+            destroyed when stop() is called.
+        """
+        self.role_maker_._barrier_worker()
+        if self.role_maker_._is_first_worker():
+            self._fleet_ptr.stop_server()
+        self.role_maker_._barrier_worker()
+        self.role_maker_._barrier_all()
+        self.role_maker_._finalize()
+
+    def distributed_optimizer(self, optimizer, strategy=None):
+        if not isinstance(optimizer, Optimizer):
+            raise ValueError("optimizer must be an instance of Optimizer")
+        self.optimizer = PSLibOptimizer(optimizer, strategy)
+        return self.optimizer
+
+    def save_inference_model(self,
+                             executor,
+                             dirname,
+                             feeded_var_names=None,
+                             target_vars=None,
+                             main_program=None,
+                             export_for_deployment=True):
+        """
+        save pserver model called from a worker
+        """
+        self._fleet_ptr.save_model(dirname)
+
+    def save_persistables(self, executor, dirname, main_program=None):
+        self._fleet_ptr.save_model(dirname)
+
+    def _set_opt_info(self, opt_info):
+        """
+        this function saves the result from DistributedOptimizer.minimize()
+        """
+        self._opt_info = opt_info
+
+
+fleet = PSLib()
+
+
+class PSLibOptimizer(DistributedOptimizer):
```

Comment:
The name of PSLibOptimizer seems wired. Could you rename this with algorithmic description to this class?